OpenRA.Mods.AS/Projectiles/CyclicImpactProjectile.cs

```diff
@@ -0,0 +1,265 @@
+
+			// target that will be assigned
+			Target target;
+
+			// subprojectiles facing
+			int facing = 0;
+
+			for (int i = 0; i < info.Offsets.Count(); i++)
+			{
+				switch (info.FireMode)
+				{
+					case FireMode.Focus:
+						offsetRotation = WRot.FromFacing(mainFacing);
+						offsetTargetPos = sourcepos + new WVec(range, 0, 0).Rotate(offsetRotation);
+						offsetSourcePos = sourcepos + info.Offsets[i].Rotate(offsetRotation);
+						break;
+					case FireMode.Line:
+						offsetRotation = WRot.FromFacing(mainFacing);
+						offsetTargetPos = sourcepos + new WVec(range + info.Offsets[i].X, info.Offsets[i].Y, info.Offsets[i].Z).Rotate(offsetRotation);
+						offsetSourcePos = sourcepos + info.Offsets[i].Rotate(offsetRotation);
+						break;
+					case FireMode.Spread:
+						offsetRotation = WRot.FromFacing(info.Offsets[i].Yaw.Facing - 64) + WRot.FromFacing(mainFacing);
+						offsetSourcePos = sourcepos + info.Offsets[i].Rotate(offsetRotation);
+						offsetTargetPos = sourcepos + new WVec(range + info.Offsets[i].X, info.Offsets[i].Y, info.Offsets[i].Z).Rotate(offsetRotation);
+						break;
+				}
+
+				if (info.Inaccuracy.Length > 0)
+				{
+					var inaccuracy = Util.ApplyPercentageModifiers(info.Inaccuracy.Length, args.InaccuracyModifiers);
+					var maxOffset = inaccuracy * (args.PassiveTarget - projectilepos).Length / range;
+					var inaccuracyOffset = WVec.FromPDF(world.SharedRandom, 2) * maxOffset / 1024;
+					offsetTargetPos += inaccuracyOffset;
+				}
+
+				target = Target.FromPos(offsetTargetPos);
+
+				// if it's true then lifespan is counted from source pos to target, instead of max range
+				lifespan = info.KillProjectilesWhenReachedTargetLocation
+					? Math.Max((args.PassiveTarget - args.Source).Length / speed.Length, 1)
+					: estimatedLifespan;
+
+				facing = (offsetTargetPos - offsetSourcePos).Yaw.Facing;
+				var projectileArgs = new ProjectileArgs
+				{
+					Weapon = args.Weapon,
+					DamageModifiers = args.DamageModifiers,
+					Facing = facing,
+					Source = offsetSourcePos,
+					SourceActor = firedBy,
+					PassiveTarget = target.CenterPosition
+				};
+
+				projectiles[i] = new CyclicImpactProjectileEffect(info, projectileArgs, lifespan, estimatedLifespan);
+				world.Add(projectiles[i]);
+			}
+		}
+
+		// gets where main projectile should fly to
+		WPos GetTargetPos()
+		{
+			var targetpos = args.PassiveTarget;
+			var actorpos = args.SourceActor.CenterPosition;
+
+			return WPos.Lerp(actorpos, targetpos, args.Weapon.Range.Length, (targetpos - actorpos).Length);
+		}
+
+		public void Tick(World world)
+		{
+			if (ticks % info.ExplosionInterval == 0 && mindelay <= ticks)
+				DoImpact();
+
+			if (ticks >= lifespan)
+			{
+				foreach (var projectile in projectiles)
+					projectile.Explode(world);
+
+				world.AddFrameEndTask(w => w.Remove(this));
+			}
+
+			ticks++;
+		}
+
+		void DoImpact()
+		{
+			// if projectile isn't dead (!DetonateSelf) then make Impact
```

Comment:
`// Trigger all so-far-untriggered explosions.` might be more accurate.  Also, comments like this one should be full sentences with a starting capital letter and fullstop at the end.